gribscan/gribscan.py

```diff
@@ -423,7 +431,13 @@ def inspect_grib_indices(messages, magician):
 
 
 def build_refs(messages, global_attrs, coords, varinfo, magician):
-    coords_inv = {k: {v: i for i, v in enumerate(vs)} for k, vs in coords.items()}
+    # xarray allows for multidimensional coordinates, for example lat/lon
+    # coordinates that change along both the x and y dimensions of a dataset.
+    # Only for the coordinates that are aligned with the data dimensions do we
+    # need to construct the inverse mapping from coordinate value to index. We
+    # select out these "principal" coordinates here.
+    principal_coords = {k: v for (k, v) in coords.items() if len(np.asarray(v).shape) == 1}
+    coords_inv = {k: {v: i for i, v in enumerate(vs)} for k, vs in principal_coords.items()}
```

Comment:
I think a slightly cleaner solution is to only create `coords_inv` for the coordinates listed in `magician.dimkeys`. This would be nicer for two reasons, _i)_ you don't need to care about your additional lat/lon coordinates, _ii)_ one would check if all required (`dimkeys`) variables are actually available in the `coords`.

Reply:
great idea! Thanks